plugin-server/src/llm-analytics/services/temporal.service.ts

```diff
@@ -0,0 +1,112 @@
+import { Client, Connection, TLSConfig } from '@temporalio/client'
+import { Counter } from 'prom-client'
+
+import { Hub } from '../../types'
+import { logger } from '../../utils/logger'
+import { captureException } from '../../utils/posthog'
+
+const temporalWorkflowsStarted = new Counter({
+    name: 'evaluation_temporal_workflows_started',
+    help: 'Number of evaluation workflows started',
+    labelNames: ['status'],
+})
+
+export class TemporalService {
+    private client?: Client
+    private connecting?: Promise<Client>
+
+    constructor(private hub: Hub) {}
+
+    private async ensureConnected(): Promise<Client> {
+        if (this.client) {
+            return this.client
+        }
+
+        if (this.connecting) {
+            return await this.connecting
+        }
+
+        this.connecting = this.createClient()
+        this.client = await this.connecting
+        this.connecting = undefined
+
+        return this.client
+    }
+
+    private async createClient(): Promise<Client> {
+        // Configure TLS if certificates are provided (for production)
+        let tls: TLSConfig | boolean = false
+        if (this.hub.TEMPORAL_CLIENT_ROOT_CA && this.hub.TEMPORAL_CLIENT_CERT && this.hub.TEMPORAL_CLIENT_KEY) {
+            tls = {
+                serverRootCACertificate: Buffer.from(this.hub.TEMPORAL_CLIENT_ROOT_CA),
+                clientCertPair: {
+                    crt: Buffer.from(this.hub.TEMPORAL_CLIENT_CERT),
+                    key: Buffer.from(this.hub.TEMPORAL_CLIENT_KEY),
+                },
+            }
+        }
+
+        // Create connection first
+        const connection = await Connection.connect({
+            address: this.hub.TEMPORAL_HOST || 'localhost:7233',
+            tls,
+        })
+
+        // Then create client with connection
+        const client = new Client({
+            connection,
+            namespace: this.hub.TEMPORAL_NAMESPACE || 'default',
+        })
+
+        logger.info('✅ Connected to Temporal', {
+            address: this.hub.TEMPORAL_HOST,
+            namespace: this.hub.TEMPORAL_NAMESPACE,
+            tlsEnabled: tls !== false,
+        })
+
+        return client
+    }
+
+    async startEvaluationWorkflow(evaluationId: string, targetEventId: string): Promise<void> {
+        try {
+            const client = await this.ensureConnected()
+
+            const workflowId = `eval-${evaluationId}-${targetEventId}-${Date.now()}`
```

Comment:
Just to keep in mind: This workflow id is more or less guaranteed to be unique as it includes the `Date.now()` part in it.

This means that if `startEvaluationWorkflow` is called multiple times with the same `evaluationId`, Temporal will not be able to identify the executions for the same `evaluationId` as duplicates, as they will all have a different workflow id.

Essentially, with a unique workflow id, is up to us to ensure workflows are actually unique. Not a bad thing at all, just to keep in mind that we are now taking up this responsibility.

Reply:
For example: with a `WorkflowIdConflictPolicy` of `USE_EXISTING` and a workflow id without a `Date.now()` we could safely call `startEvaluationWorkflow` as many times as we want with the same `evaluationId`. Temporal would only ever create just one execution per workflow id. All the duplicate calls would just return the handle for the first execution created.

This is a neat feature to ensure idempotency if this is a concern!